go-controller/pkg/ovn/hybrid.go

```diff
@@ -314,16 +314,19 @@ func (oc *DefaultNetworkController) removeHybridLRPolicySharedGW(node *kapi.Node
 		return fmt.Errorf("failed to delete policy %s from %s, error: %w", name, ovntypes.OVNClusterRouter, err)
 	}
 
-	if err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(oc.nbClient, ovntypes.OVNClusterRouter, func(item *nbdb.LogicalRouterStaticRoute) bool {
-		return strings.Contains(item.ExternalIDs["name"], name)
-	}); err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
-		return fmt.Errorf("failed to delete static route %s from %s, error: %w", name, ovntypes.OVNClusterRouter, err)
-	}
+	// order is important here to not incur referential integrity violation
```

Comment:
I'm not understanding how removing routes on 2 different routers have anything to do with each other.  What am I missing?

Reply:
So the first op to remove the routes from `ovntypes.OVNClusterRouter` has a predicate that also matches and removes the routes referenced from `ovntypes.GWRouterPrefix+nodeName` without removing the references themselves and the transaction fails because those latter references point to non-existing routes.

There are several options here, from using a more specific predicate on the first op (@JacobTanenbaum would that be feasible?) to use the same transaction for both ops which I didn't go for because I was working in a context of bigger changes and did not want to over-complicate it. But now that this sits on its own PR we can probably do whatever we find its best.

Reply:
Another option is to change the op to just remove the references and not remove the routes and let garbage collection do its thing. I had the idea on mind of reviewing the ops once we have referential integrity in place in the test server to simplify a lot of the ops in this way.

Reply:
Oh I think that is what I was missing...I didn't realize we were also *deleting the routes* and not just removing them from the router and letting the DB clean them up. I think we should do this instead.

Reply:
Can we have this stop gap solution or any of the other alternatives until we have referential integrity in for the tests? Otherwise the tests are going to require additional work that we will have to undo when we add referential integrity

Reply:
I think this stop gap would be acceptable for now. I think I would prefer the option of changing the op to just remove references and changing the name if possible to indicate what it is actually doing. 

Could you more clearly indicate in the comment and create an upstream issue for the work needed to be done?

Reply:
@JacobTanenbaum ah probably not, it seems that the name can be different...few lines above:
```
name := ovntypes.HybridSubnetPrefix + nodeName + ":" + hoNodeName
if hoNodeName == hybridCidrClusterKey {
name = ovntypes.HybridSubnetPrefix + nodeName
}
```
changed in
https://github.com/ovn-org/ovn-kubernetes/commit/2d44a24dd652b1a5003cdd29dcb3cf4b92562e12
which was what probably introduced the issue.

So I would probably keep the current approach OR bundle all the ops in the same transaction for now. @trozet let me know what you think.

Reply:
Here we have it, assigned to myself: https://github.com/ovn-org/ovn-kubernetes/issues/4089